### ISSUE TITLE
Fork of working version

### ISSUE THREAD

**Issue by akim-mcmath** [DanielCaspers/gulp-flat#1]
Hi,

Since you [invited contributions](http://stackoverflow.com/questions/35564976/nodejs-gulp-streams-vinyl-file-objects-gulp-wrapper-for-npm-package-produci), I decided I would give it a shot. You can view my fork of the repo [here](https://github.com/akim-mcmath/gulp-flat). Feel free to look it over, clone it, or whatever you like. If you would like, I can submit a pull request. If it's not to your liking, don't worry about it. But I thought I would try an implementation of the project, and hopefully it will at least give you some ideas.

Here are the changes that I made:

* Fixed the issue with buffered mode, so that the output displays correctly.
* Added support for streaming mode.
* Omitted [through2](https://github.com/rvagg/through2) as a dependency, since it lacks support for
utilizing the [Transform Stream's](https://nodejs.org/api/stream.html#stream_class_stream_transform_1) contstructor, which was necessary to implement
the streaming mode version.
* Added tests using [mocha](https://mochajs.org/) and [chai](http://chaijs.com/api/).
* Added coverage support using [Istanbul](https://github.com/gotwarlost/istanbul).
* Added linting support using [JSHint](http://jshint.com/about/).
* Added configuration files: [.editorconfig](http://editorconfig.org/), .gitignore, .jshintrc.
* Prepared repo for continuous intigration with [Travis-CI](https://travis-ci.org/) by including a .travis.yml config file
*Prepared repo for coverage reporting with [Coveralls](https://coveralls.io/), if you would like
to display the relevant badges.
* Updated README with API section.
* Updated package.json with scripts, dependencies, and other relevant
   fields.

The following tasks still need to be done:

* If you want to display CI and coverage badges, you should
add this repo to your Travis-CI and Coveralls accounts. The repo is
already configured for those services.
* You should also update the README with installation instructions, perhaps
an example or two, and any badges you want to display.

**Comment by DanielCaspers** [comment]
Hi Akim,

I will definitely be looking over this more closely tomorrow (which is why
I haven't marked the post as resolved yet). I think the fact that you went
this far is just stellar! I really appreciate that you helped this along.
In fact, my original contributions seem somewhat insignificant in
comparison now in my opinion, but it at least it was a learning experience
too.

Also, if I decide to go the pull request route, what are your thoughts on
me rebasing some of this and expanding the commits? My initial thought is
that if I don't want to pull in all of the changes (for example, if some of
the tools violate the flat license (which I am actually currently in
violation of and must change), would you be ok with the work being
separated out so that the majority gets in, but some unique features stay
in your repo?

What are your thoughts on co-authoring this on NPM once we both deem it
worthy of posting? I would be more than happy to co-author this with you if
you would accept!

Sincerely,
Daniel Caspers


On Mon, Feb 22, 2016 at 9:32 PM, Akim McMath <notifications@github.com>
wrote:

> Hi,
>
> Since you invited contributions
> <http://stackoverflow.com/questions/35564976/nodejs-gulp-streams-vinyl-file-objects-gulp-wrapper-for-npm-package-produci>,
> I decided I would give it a shot. You can view my fork of the repo here
> <https://github.com/akim-mcmath/gulp-flat>. Feel free to look it over,
> clone it, or whatever you like. If you would like, I can submit a pull
> request. If it's not to your liking, don't worry about it. But I thought I
> would try an implementation of the project, and hopefully it will at least
> give you some ideas.
>
> Here are the changes that I made:
>
>    - Fixed the issue with buffered mode, so that the output displays
>    correctly.
>    - Added support for streaming mode.
>    - Omitted through2 <https://github.com/rvagg/through2> as a
>    dependency, since it lacks support for utilizing the Transform Stream's
>    <https://nodejs.org/api/stream.html#stream_class_stream_transform_1>
>    contstructor, which was necessary to implement the streaming mode version.
>    - Added tests using mocha <https://mochajs.org/> and chai
>    <http://chaijs.com/api/>.
>    - Added coverage support using Istanbul
>    <https://github.com/gotwarlost/istanbul>.
>    - Added linting support using JSHint <http://jshint.com/about/>.
>    - Added configuration files: .editorconfig <http://editorconfig.org/>,
>    .gitignore, .jshintrc.
>    - Prepared repo for continuous intigration with Travis-CI
>    <https://travis-ci.org/> by including a .travis.yml config file
>    *Prepared repo for coverage reporting with Coveralls
>    <https://coveralls.io/>, if you would like to display the relevant
>    badges.
>    - Updated README with API section.
>    - Updated package.json with scripts, dependencies, and other relevant
>    fields.
>
> The following tasks still need to be done:
>
>    - If you want to display CI and coverage badges, you should add this
>    repo to your Travis-CI and Coveralls accounts. The repo is already
>    configured for those services.
>    - You should also update the README with installation instructions,
>    perhaps an example or two, and any badges you want to display.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DanielCaspers/gulp-flat/issues/1>.
>



-- 
Dan Caspers
Keyboardist for LikeABossa <http://likeabossa.com/>
952-270-6622 <+19522706622> | LinkedIn Profile
<https://www.linkedin.com/in/danielcaspers>

**Comment by akim-mcmath** [comment]
Yes, I thought it was meant to be a learning experience in part, so I thought I would just show you how I would go about it and you can take from it what you wish. In fact, it's a bit of a learning experience for me too; I only wrote my first NPM package a few months ago. I'm glad you're pleased with it anyway.

You are welcome to expand on the commits, or whatever you like. In face, you could just copy the bits that you like without worrying about a pull request (this might be better for you from a learning-experience perspective). Don't worry about listing me as a co-author. I'm happy to be listed as a contributor, since it was your idea in the first place. And I don't really want to have to worry about maintaining it, so if you decide to do that, you'll end up doing a lot more work than I did.

I'm not sure what you mean about violating the [flat license](https://github.com/hughsk/flat/blob/master/LICENSE). It's open source and licensed under BSD-3, unless I'm missing something. I don't think there's anything in either of our repos that would violate the license.
